dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt

```diff
@@ -256,6 +258,30 @@ object Datadog {
         }
     }
 
+    // Executes all the pending queues in the upload/persistence executors.
+    // Tries to send all the granted data for each feature and then clears the folders.
+    // You should not use this method in production code. This is mainly for test purposes.
+    @Suppress("unused")
+    private fun flushStoredData() {
+        if (initialized.get()) {
+            val tasks = ArrayList<Runnable>()
+            (CoreFeature.persistenceExecutorService as? ThreadPoolExecutor)
+                ?.queue
+                ?.drainTo(tasks)
+            // we make sure we upload the currently locked files
+            CoreFeature.uploadExecutorService
+                .queue
+                .drainTo(tasks)
+            tasks.forEach {
+                it.run()
+            }
```

Comment:
because you run the Upload tasks, you need to make sure the newly scheduled upload tasks won't be ran later. Maybe after draining you can disable the executor service to prevent any new task to be scheduled

Reply:
Like I explained above to @0xnm 's comment this method will be called in the test rule after the process has been idled. During my tests this scenario did not occur and my intention was to actually stop the executors here but then this will not be anymore a flush method only, it will have mixed capabilities. I would rather keep it like this and in case this is needed I will add it later